src/api/handlers/webhook.ts

```diff
@@ -281,8 +281,7 @@ export async function deleteWebhookMessage(
  * Guild commands update **instantly**. We recommend you use guild commands for quick testing, and global commands when they're ready for public use.
  */
 export async function createSlashCommand(options: CreateSlashCommandOptions) {
-  // Use ... for content length due to unicode characters and js .length handling
-  if ([...options.name].length < 2 || [...options.name].length > 32) {
+  if (/^[\w-]{1,32}$/.test(options.name)) {
```

Comment:
We need to check if the name does not match the regex.
```suggestion
  if (!/^[\w-]{1,32}$/.test(options.name)) {
```